### PR TITLE
feature to allow to link two extraction graphs

### PR DESCRIPTION
**Pull request by maxkozlovsky**
Feature to support making an output of an extraction graph an input to another extraction graph. All top level policies (with content source ingestion) of the linked graph are applied to the specified graph/content_source combination.

### CODE REVIEW

File: src/api.rs [hunk 1]
```diff
@@ -15,6 +15,13 @@ use utoipa::{IntoParams, ToSchema};
 
 use crate::{api_utils, metadata_storage, vectordbs};
 
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
+pub struct ExtractionGraphLink {
+    pub source_graph_name: String,
+    pub content_source: String,
+    pub linked_graph_name: String,
```
Comment (diptanu) [inline src/api.rs:22]: Are there any checks to prevent two graphs from pointing to each other?

**Review by diptanu — CHANGES_REQUESTED** [review]
Left some comments, looking good otherwise.

File: src/state/store/state_machine_objects.rs [hunk 2]
```diff
@@ -698,6 +700,9 @@ pub struct IndexifyState {
 
     /// Next change id
     pub change_id: std::sync::Mutex<u64>,
+
+    /// Graph+ContentSource->Policy Id
+    pub graph_links: Arc<RwLock<HashMap<ExtractionGraphNode, HashSet<String>>>>,
```
[the remark this answers is not in the document]
Comment (maxkozlovsky) [reply src/state/store/state_machine_objects.rs]: ok

File: src/api.rs (hunk 1 above)
Comment (maxkozlovsky) [reply src/api.rs:22]: ok

**Pull request merged**